libavcodec/jpeg2000dec.c

```diff
@@ -54,6 +54,9 @@
 #define HAD_COC 0x01
 #define HAD_QCC 0x02
 
+// Pseudo mode for placeholder passes
```

Comment:
“Pseudo mode” is a coding mode for handling placeholder passes. It works with the HT mode (0x40), and the parser assumes all HT code block has placeholder passes ( |= HT_PLHD). This flag is canceled once an HT segment with a non-zero length is found. Because SPcod is limited to 8 bits, this flag (beyond 8 bits)  cannot exist in a standardized way.

Reply:
Ah, so this is to keep state using an existing field/variable?

Reply:
Specifically, why not add a state variable in `Jpeg2000Cblk`?

Reply:
It has been added. Please see https://github.com/sandflow/ffmpeg-ht/blob/1ba6b4d95d857727912a5ee0868128f67acf222a/libavcodec/jpeg2000.h#L195

Adding a dedicated flag is another option. What do you think?

Reply:
I think a dedicated flag has advantages:

- it avoids changing an existing field
- it avoids diverging from the semantics specified in the standard
- it clearly separates HT from non-HT state (maybe not a great advantage)

Reply:
I agree. In the next commit, I’ll modify the code to use a dedicated flag.